services/job_runner.go

```diff
@@ -34,6 +34,11 @@ func BeginRunAtBlock(
 	if err != nil {
 		return models.JobRun{}, err
 	}
+	if input.Amount != nil &&
+		store.Config.MinimumContractPayment.Cmp(input.Amount) > 0 {
+		logger.Infow(fmt.Sprintf("Rejecting job %s with payment %s below minimum threshold (%s)", job.ID, input.Amount, store.Config.MinimumContractPayment.Text(10)))
```

Comment:
Yeah, that's the first check mark above - I feel like they should all treat the errors the same and use `ApplyStatus`. Thoughts?

Reply:
I'm all for treating the errors the same. Let's get @se3000 to chime in. Is `ApplyStatus` a new thing? We have `WithError` because it needs an error message to complement the StatusErrored.

Reply:
The tricky thing about treating the error the same way is that the errors around endAt and startAt are expected. Our cron process [keeps running jobs after the endAt has passed](https://www.pivotaltracker.com/story/show/154424737). In the cron scenario, we want to silently ignore the errors because they're expected. In the payment scenario, I think we should create a record of a run request being ignored.

Two options are immediately clear to me:
1) Handle the errors in different places so that the different types of errors can be handled differently.
2) Have some kind of flag on the error so that we can differentiate if it's an expected error on our part(cron), or an error on the requester's part(payment).

I'm currently leaning towards 2, because I like the idea of having all error handling in one place, but it essentially means we're doing type checking on errors. Something I'm missing?

Reply:
Great point about the cron process. Right now, we handle it as you describe in part 2: startAt,endAt are explicitly returned an error in the tuple before ExecuteRun, noMoney() sets an error on the job run and let's it continue.

For the purposes of this PR, should we leave it as is and create another story to better handle this scenario?

Reply:
Works for me. We'll be attaching payment to other initiators, and will have more information on how to design this when we get there.